web/models/announcement.ex

```diff
@@ -44,6 +44,12 @@ defmodule Constable.Announcement do
     ]
   end
 
+  def without_interests(query, interests \\ []) do
+    from a in query,
+      full_join: i in assoc(a, :interests),
+      where: not i.name in ^interests or is_nil(i.name)
```

Comment:
Do we want to filter announcements with no interests though? It seems like that might be something we want still since sometimes people forget to add them. Other than that I think the query looks pretty darn good!